File: src/Stronghold.Application/Features/Seminars/SeminarResponse.cs
namespace Stronghold.Application.Features.Seminars;

public class SeminarResponse
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Lecturer { get; set; } = string.Empty;
    public DateTime StartDate { get; set; }
    public int DurationMinutes { get; set; }
    public int MaxCapacity { get; set; }
    public int RegisteredCount { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class SeminarRegistrationResponse
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string UserFullName { get; set; } = string.Empty;
    public string UserEmail { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
}

File: stronghold_desktop/lib/features/seminars/models/seminar_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'seminar_response.freezed.dart';
part 'seminar_response.g.dart';

@freezed
abstract class SeminarResponse with _$SeminarResponse {
  const factory SeminarResponse({
    required int id,
    required String name,
    required String description,
    required String lecturer,
    required DateTime startDate,
    required int durationMinutes,
    required int maxCapacity,
    required int registeredCount,
    required DateTime createdAt,
  }) = _SeminarResponse;

  factory SeminarResponse.fromJson(Map<String, dynamic> json) =>
      _$SeminarResponseFromJson(json);
}

@freezed
abstract class PagedSeminarResponse with _$PagedSeminarResponse {
  const factory PagedSeminarResponse({
    required List<SeminarResponse> items,
    required int totalCount,
    required int totalPages,
    required int currentPage,
    required int pageSize,
  }) = _PagedSeminarResponse;

  factory PagedSeminarResponse.fromJson(Map<String, dynamic> json) =>
      _$PagedSeminarResponseFromJson(json);
}

@freezed
abstract class SeminarRegistrationResponse with _$SeminarRegistrationResponse {
  const factory SeminarRegistrationResponse({
    required int id,
    required int userId,
    required String userFullName,
    required String userEmail,
    required DateTime createdAt,
  }) = _SeminarRegistrationResponse;

  factory SeminarRegistrationResponse.fromJson(Map<String, dynamic> json) =>
      _$SeminarRegistrationResponseFromJson(json);
}
